feat: streamline running e2e testing locally

vcastellm opened the pull request:
This pull request adds a new utility script to help synchronize the `aggkit-config.toml` configuration file between the local test environment and the Kurtosis CDK repository. The script provides clear instructions, error handling, and user prompts to ensure the configuration stays in sync.

New script for config synchronization:

* Added `test/kurtosis-config.sh`, a Bash script that:
  - Syncs `aggkit-config.toml` between the local `test/config` directory and the Kurtosis CDK repo template.
  - Provides usage instructions and examples for ease of use.
  - Handles both directions: pushes local changes to the repo or pulls the template if the local copy is missing.
  - Includes informative colored logging and robust error handling.
  - Check the commit id in Kurtosis repo and allow the user to update it

![Kapture 2025-11-27 at 13 04 31](https://github.com/user-attachments/assets/4041273f-13e6-485a-b314-34830c7d1113)

  What Changed

  1. Optional Repository Paths

  - Both kurtosis_repo_path and e2e_repo_path parameters are now optional
  - Only test_type is required
  - Usage: ./run-local-e2e.sh <test_type> [kurtosis_repo_path] [e2e_repo_path]

  2. Automatic Cloning with User Prompts

  - When a repo path is not provided, the script prompts: "Do you want to clone the [repo name] temporarily? [Y/n]"
  - If user confirms (default is Yes), the repo is cloned to a temporary directory
  - If user declines, that step is skipped

  3. Repository URLs

  - Kurtosis CDK: https://github.com/0xPolygon/kurtosis-cdk.git
  - E2E tests: https://github.com/agglayer/e2e.git

  4. Automatic Cleanup

  - Temporary clones are automatically cleaned up on script exit (success or failure)
  - Uses trap handlers to ensure cleanup happens

  5. Smart Commit Verification

  - Temporary Kurtosis clones are checked out to the expected commit automatically
  - User-provided Kurtosis repos are verified and user is prompted to checkout the correct commit if needed

  6. Config File Syncing

  - If test/config/aggkit-config.toml exists locally, it's automatically synced to the Kurtosis template

  Usage Examples

  ### Prompt to clone both repos temporarily
  `./test/run-local-e2e.sh single-l2-network-op-succinct`

  ### Use existing Kurtosis, prompt for E2E
  `./test/run-local-e2e.sh single-l2-network-op-succinct /path/to/kurtosis-cdk`

  ### Use both existing repos
  `./test/run-local-e2e.sh single-l2-network-op-succinct /path/to/kurtosis-cdk /path/to/e2e`

  ### Skip Kurtosis, use existing E2E
  `./test/run-local-e2e.sh single-l2-network-op-succinct - /path/to/e2e`

  ### Use Kurtosis, skip E2E
  `./test/run-local-e2e.sh single-l2-network-op-succinct /path/to/kurtosis-cdk -`

```
Usage: test/run-local-e2e.sh [OPTIONS] <test_type> [kurtosis_repo_path] [e2e_repo_path]

Options:
  -f, --force         Automatically clone repos without prompting (non-interactive mode)
  -h, --help          Show this help message

Arguments:
  test_type           Type of test to run (required)
                      Options: single-l2-network-op-succinct
                               single-l2-network-op-succinct-aggoracle-committee
                               single-l2-network-op-pessimistic
                               multi-l2-networks-2-chains-op-pessimistic
                               multi-l2-networks-3-chains-cdk-erigon-pessimistic
  kurtosis_repo_path  Path to Kurtosis CDK repo (optional)
                      - If not provided: Will prompt to clone temporarily (or auto-clone with --force)
                      - Use '-' to skip Kurtosis setup entirely
  e2e_repo_path       Path to E2E repo (optional)
                      - If not provided: Will prompt to clone temporarily (or auto-clone with --force)
                      - Use '-' to skip E2E tests entirely

Examples:
  test/run-local-e2e.sh single-l2-network-op-succinct                                     # Prompt to clone both repos
  test/run-local-e2e.sh -f single-l2-network-op-succinct                                  # Auto-clone both repos (no prompts)
  test/run-local-e2e.sh single-l2-network-op-succinct /path/to/kurtosis /path/to/e2e     # Use existing repos
  test/run-local-e2e.sh --force single-l2-network-op-succinct /path/to/kurtosis          # Use Kurtosis, auto-clone E2E
  test/run-local-e2e.sh single-l2-network-op-succinct - /path/to/e2e                     # Skip Kurtosis, use E2E repo
  test/run-local-e2e.sh single-l2-network-op-succinct /path/to/kurtosis -                # Use Kurtosis, skip E2E
```

Stefan-Ethernal committed:
chore: sync main with develop (#203)

cypherpepe committed:
Fixed dead link in `aggoracle.md`

Stefan-Ethernal committed:
chore: fixed dead link in `aggoracle.md` (#376)

vcastellm committed:
chore: add AGENTS.md for build, style, structure

vcastellm committed:
test: add test script to sync aggkit-config.toml

vcastellm committed:
Read Kurtosis commit from workflow and verify repo

vcastellm committed:
chore: rename sync-aggkit-config.sh to kurtosis-config.sh

Commit:
Merge branch 'develop' into inject_config

Commit:
Merge branch 'develop' into inject_config

Commit:
Merge branch 'develop' into inject_config

joanestebanr reviewed (commented):
- Maybe make sense to allow to specify which config-file to use instead the same `aggkit-config.toml` always
- About kurtosis-cdk commit... maybe would be useful to clone a repo if you don't pass a path (so you clone it to use just in that moment). Even it can be used into CI to be more close to run locally
- The script `run-local-e2e.sh` could call to the this script to verify the kurtosis-cdk

WDYT?

vcastellm committed:
Determine kurtosis-cdk ref and checkout repo

Use the native checkout action instead of custom code and API calls

vcastellm committed:
Support optional Kurtosis repo path and auto-clone

Add cleanup trap to remove temporary clone on exit Clone kurtosis-cdk if
no path provided and checkout expected commit

vcastellm committed:
Merge remote-tracking branch 'refs/remotes/origin/inject_config' into inject_config

vcastellm committed:
Verify kurtosis-cdk commit in e2e script

Use test/kurtosis-config.sh to verify the provided kurtosis-cdk repo is
on the expected commit before starting the local e2e run.

vcastellm committed:
Add temporary cloning and cleanup to local E2E script

vcastellm committed:
Merge branch 'main' into inject_config

vcastellm committed:
Merge branch 'develop' into inject_config

vcastellm committed:
Restore doc

vcastellm committed:
Bump kurtosis-cdk

joanestebanr reviewed (commented):
I suggest to add a flag `-y` to avoid the question about: 
|  When a repo path is not provided, the script prompts: "Do you want to clone the [repo name] temporarily? [Y/n]"

So, if you known that you want to clone it pass in the command line, WDYT?

vcastellm commented:
> I suggest to add a flag `-y` to avoid the question about: | When a repo path is not provided, the script prompts: "Do you want to clone the [repo name] temporarily? [Y/n]"
> 
> So, if you known that you want to clone it pass in the command line, WDYT?

Traditionally you use `yes | command` in linux for non-interactive "yes", right? Also it's the default option so it's just pressing enter, IMO it's redundant.

Edit: Added `--force` as per @joanestebanr suggestion

vcastellm committed:
Send log output to stderr

vcastellm committed:
Add --force flag and non-interactive cloning

Commit:
Merge branch 'develop' into inject_config

Commit:
Merge branch 'develop' into inject_config

Commit:
Merge branch 'develop' into inject_config

Commit:
Merge branch 'develop' into inject_config

vcastellm committed:
revert indentation

vcastellm committed:
Simplify test-e2e workflow and update CDK commit
